vault/expiration.go

```diff
@@ -51,11 +51,13 @@ const (
 // If a secret is not renewed in timely manner, it may be expired, and
 // the ExpirationManager will handle doing automatic revocation.
 type ExpirationManager struct {
-	router     *Router
-	idView     *BarrierView
-	tokenView  *BarrierView
-	tokenStore *TokenStore
-	logger     log.Logger
+	core          *Core
+	router        *Router
+	idView        *BarrierView
+	tokenView     *BarrierView
+	tokenStore    *TokenStore
+	logger        log.Logger
+	coreStateLock *sync.RWMutex
```

Comment:
If we are storing `core`, we don't need to also store the state lock

Reply:
It's true, the main reason I kept it like this is I would really rather not plumb through core, I just also figured we should check standby and sealed while we were at it and didn't feel like storing two pointers to bools. So I left it the easiest way to plumb back out.